internal/querycoordv2/meta/resource_manager.go

```diff
@@ -576,7 +577,7 @@ func (rm *ResourceManager) selectMissingRecoverSourceRG(rg *ResourceGroup) *Reso
 	// First, Transfer node from most redundant resource group first. `len(nodes) > limits`
 	if redundantRG := rm.findMaxRGWithGivenFilter(
 		func(sourceRG *ResourceGroup) bool {
-			return rg.GetName() != sourceRG.GetName() && sourceRG.RedundantNumOfNodes() > 0
+			return rg.GetName() != sourceRG.GetName() && sourceRG.RedundantNumOfNodes() > 0 && rg.AcceptNode(sourceRG.PreferRemovedNode())
```

Comment:
Here could be a failure logic.
In current implementation, If the rg1's `PerferRemoveNode()` return a node that  rg2 don't accept, the rg2 can never get node from the rg1. 
So the PreferRemoveNode should be a logic that passing `rg1` as a argument to determine the best and stable node that move from `rg2` into `rg1`.

Reply:
I prefer that the `selectMissingRecoverSourceRG` return the sourceRG and the target node id together to avoid unstable node selection